lib/facility.rb

```diff
@@ -1,14 +1,62 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader   :name,
+                :address,
+                :phone,
+                :services,
+                :registered_vehicles,
+                :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
+    @services = @services.uniq
+  end
+
+  def vehicle_type(vehicle)
+    return :ev if vehicle.electric_vehicle?
```

Comment:
Guard statements work here, an ifelse could work too

Reply:
idk why I hate if/else so much. Must be something from my childhood.